src/app/client/src/app/modules/shared-feature/components/user-location/user-location.component.ts

```diff
@@ -294,7 +289,61 @@ export class UserLocationComponent implements OnInit {
       locationDetails.districtCode = this.userDetailsForm.value.district;
     }
     const data = {locationCodes: locationCodes};
-    this.updateLocation(data, locationDetails);
+    let districtData, stateData, changeType = '';
+    if (locationDetails.stateCode) {
+      stateData = _.find(this.allStates, (states) => {
+        return states.code === locationDetails.stateCode;
+      });
+    }
+    if (locationDetails.districtCode) {
+      districtData = _.find(this.allDistricts, (districts) => {
+        return districts.code === locationDetails.districtCode;
+      });
+    }
+    if (stateData.name !== _.get(this.suggestedLocation, 'state.name')) {
+      changeType = changeType + 'state-changed';
```

Comment:
please change so that the value is "state-dist-changed" not "state-changed-dist-changed"